src/app/key/key.component.html

```diff
@@ -1,6 +1,6 @@
 <div class="key" id="key-{{key}}">
-    <button [ngClass]="isLargeButton(key)
-    ? 'large-btn'
-    : 'small-btn' "
-    (click)="onKeyClick(key)">{{key}}</button>
-</div>
+    <button [class.light-grey]="(keyColor$| async) === 'light-grey'"
```

Comment:
I'm imagining a scenario where your UX team decides overhaul the existing different color scheme and replace the existing colors with new ones. I think currently this would require changing each color name in 3 or 4 different places. Is there a way to refactor so that you make this code easier to maintain?

I'd think about how the list of possible states would be less likely to change than the list of associated colors. In our codebase we sometimes use typescript Records to associate enum types representing a fixed list of different states with an associated color or style.

Reply:
The first iteration of the `KeyColor` enum looked like:
```
enum KeyColor {
   darkGrey = '#787c7f',
   lightGrey = '#EFEFEF',
   green = '#6ca965',
   yellow = '#c8b653'
}
```
But I ran into the blocker I posted about yesterday when trying to apply the enum values to `[style.background-color]`, partially because it was trying to set the background-color before the Observable was done emitting a value. But I also wasn't using the async pipe... would something like,`[style.background-color]="(keyColor$| async)"`, where `keyColor$` is the hex value, work in this instance?

Reply:
What if, instead of having colors be your key, you made a KeyState enum and set each of them equal to a different classname string? That way you could provide the background color for each class in your css file & that would be the sole reference for the color that each state should be associated with

Reply:
Oooh I'm here now! Love that idea. Honestly it could be used for the board tiles as well!

Reply:
Another thing I just remembered: you can also make it simpler by just using the async pipe with your observable in the *ngIf conditional. Sometimes we have to wait to render based on a certain property of a complex object observable, in which makes sense to subscribe to it & evaluate each emitted value. But in your case you may not need to

Either way, I'd try to approach your refactor in a way where the component is depending on states/classes & not so tightly coupled to colors

Reply:
Ok I just pushed up some changes that I think match what you were thinking, go ahead and take a look and let me know your thoughts!

Reply:
I didn't end up using `*ngIf` but I did get rid of `keyColor$` all together and used an async pipe like you suggested!